#!/usr/bin/env python
PACKAGE = "hik_ros1_camera_driver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("exposure_auto",   bool_t,   0, "Exposure auto parameter",  False)
gen.add("exposure_time",    int_t,    0, "Exposure time parameter", 5000,  0, 200000)
gen.add("gain_auto",   bool_t,   0, "Gain auto parameter",  False)
gen.add("gain",    int_t,    0, "Gain parameter", 0,  0, 50)
gen.add("gamma_enable",   bool_t,   0, "Gamma enable parameter",  True)
gen.add("gamma",    int_t,    0, "Gamma parameter", 0,  0, 50)

exit(gen.generate(PACKAGE, "hik_ros1_camera_driver", "CameraDynReconf"))